lib/internal/child_process.js

```diff
@@ -655,6 +655,16 @@ function setupChannel(target, channel) {
     if (message === undefined)
       throw new ERR_MISSING_ARGS('message');
 
+    // Non-serializable messages should not reach the remote
+    // end point; as any failure in the stringification there
+    // will result in error message that is weakly consumable.
+    // So perform a sanity check on message prior to sending.
+    const ret = JSON.stringify(message);
+    if (ret == null) {
```

Comment:
Is not `null` serializable?
```
> JSON.parse(JSON.stringify(null))
null
```

Reply:
Maybe `=== undefined`?

Reply:
With example in the issue, if we go with:
```js
subprocess.send(null);
```
we currently get:
```
SUBPROCESS got message: null
```
So this change can be breaking.

Reply:
BTW, is sending `null` to subprocess tested in our suite?

Reply:
```
> JSON.stringify(null) == null
false
> JSON.stringify(null) == 'null'
true
> 
```

That does not seem to break anything, as stringifier on `null` returns a string 'null' as opposed to `null` - though visibly this can be confusing.

If you have strong opinions I can make changes at my side by confining it to `undefined`, but to be strict with non-serializable objects, the current logic may be better?

Reply:
You are right. Sorry, I was confused.